`403 Forbidden` error when updating existing `pagerduty_escalation_policy`'s target

We are experiencing `403 Forbidden` error when attempting to modify existing `pagerduty_escalation_policy`'s target. 

### Terraform Version

```bash
Terraform v1.6.6
on darwin_arm64
+ provider registry.terraform.io/pagerduty/pagerduty v3.7.1
```

### Affected Resource(s)

- `pagerduty_escalation_policy`

### Terraform Configuration Files
```hcl
resource "pagerduty_escalation_policy" "team_oncall_escalation_policy" {
  name      = "${var.team_name} Escalation Policy"
  num_loops = 9

  dynamic "rule" {
    for_each = var.team_oncall_enabled == true ? [1] : []
    content {
      escalation_delay_in_minutes = 5
      target {
        type = "schedule_reference"
        id   = pagerduty_schedule.team_oncall_schedule_primary.id
      }
    }
  }

  dynamic "rule" {
    for_each = var.team_oncall_enabled == true ? [1] : []
    content {
      escalation_delay_in_minutes = 5
      target {
        type = "schedule_reference"
        id   = pagerduty_schedule.team_oncall_schedule_secondary.id
      }
    }
  }

  rule {
    escalation_delay_in_minutes = 60
    target {
      type = "schedule_reference"
      id   = var.oncall_schedule_id
    }
  }

  rule {
    escalation_delay_in_minutes = 15
    target {
      type = "schedule_reference"
      id   = var.manager_schedule_id
    }
  }

  lifecycle {
    create_before_destroy = true
  }
}
```

```bash
Terraform will perform the following actions:

  # pagerduty_escalation_policy.team_oncall_escalation_policy will be updated in-place
  ~ resource "pagerduty_escalation_policy" "team_oncall_escalation_policy" {
        id          = "PJ81HJ6"
        name        = "<omitted> Escalation Policy"
        # (3 unchanged attributes hidden)

      ~ rule {
            id                          = "<omitted>"
            # (1 unchanged attribute hidden)

          ~ target {
              ~ id   = "PAIH89Y" -> "PZ4PJ9I"
                # (1 unchanged attribute hidden)
            }

            # (1 unchanged block hidden)
        }

        # (3 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

### Debug Output

```bash
2024-02-15T13:06:51.880+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: 2024/02/15 13:06:51 ===== PagerDuty Cache Skipping Init =====
2024-02-15T13:06:52.252+0100 [INFO]  provider.terraform-provider-pagerduty_v3.7.1: [INFO] PagerDuty client configured
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_v is [map[type:assign_to_everyone]]
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_teras is "assign_to_everyone"
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_v is [map[type:assign_to_everyone]]
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_teras is "assign_to_everyone"
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_v is [map[type:assign_to_everyone]]
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_teras is "assign_to_everyone"
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_v is [map[type:assign_to_everyone]]
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: expandEscalationRuleAssignmentStrategy_teras is "assign_to_everyone"
2024-02-15T13:06:52.252+0100 [INFO]  provider.terraform-provider-pagerduty_v3.7.1: [INFO] Updating PagerDuty escalation policy: <omitted>
2024-02-15T13:06:52.252+0100 [DEBUG] provider.terraform-provider-pagerduty_v3.7.1: [DEBUG] Waiting for state to become: [success]
2024-02-15T13:06:52.502+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 500ms before next try
2024-02-15T13:06:53.277+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 1s before next try
2024-02-15T13:06:54.539+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 2s before next try
2024-02-15T13:06:56.780+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 4s before next try
2024-02-15T13:07:01.037+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 8s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 10s elapsed]
2024-02-15T13:07:09.305+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 20s elapsed]
2024-02-15T13:07:19.569+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 30s elapsed]
2024-02-15T13:07:29.813+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 40s elapsed]
2024-02-15T13:07:40.065+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 50s elapsed]
<omitted for brevity>
2024-02-15T13:11:35.749+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 4m50s elapsed]
2024-02-15T13:11:45.990+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: [TRACE] Waiting 10s before next try
.pagerduty_escalation_policy.team_oncall_escalation_policy: Still modifying... [id=PJ81HJ6, 5m0s elapsed]
2024-02-15T13:11:52.255+0100 [WARN]  provider.terraform-provider-pagerduty_v3.7.1: [WARN] WaitForState timeout after 5m0s
2024-02-15T13:11:52.255+0100 [WARN]  provider.terraform-provider-pagerduty_v3.7.1: [WARN] WaitForState starting 30s refresh grace period
2024-02-15T13:11:54.257+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: Called downstream: @module=sdk.helper_schema tf_mux_provider="*schema.GRPCProviderServer" tf_provider_addr=registry.terraform.io/pagerduty/pagerduty @caller=github.com/hashicorp/terraform-plugin-sdk/v2@v2.31.0/helper/schema/resource.go:920 tf_req_id=bf5f7fd9-6e12-dd97-4231-b2f04f3b62f8 tf_resource_type=pagerduty_escalation_policy tf_rpc=ApplyResourceChange timestamp="2024-02-15T13:11:54.256+0100"
2024-02-15T13:11:54.267+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: Received downstream response: tf_proto_version=5.4 tf_req_duration_ms=302378 tf_req_id=bf5f7fd9-6e12-dd97-4231-b2f04f3b62f8 tf_rpc=ApplyResourceChange diagnostic_error_count=1 diagnostic_warning_count=0 tf_provider_addr=registry.terraform.io/pagerduty/pagerduty @caller=github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/internal/tf5serverlogging/downstream_request.go:40 @module=sdk.proto tf_resource_type=pagerduty_escalation_policy timestamp="2024-02-15T13:11:54.258+0100"
2024-02-15T13:11:54.267+0100 [ERROR] provider.terraform-provider-pagerduty_v3.7.1: Response contains error diagnostic: @module=sdk.proto diagnostic_summary="PUT API call to https://api.pagerduty.com/escalation_policies/PJ81HJ6 failed: 403 Forbidden" tf_provider_addr=registry.terraform.io/pagerduty/pagerduty tf_req_id=bf5f7fd9-6e12-dd97-4231-b2f04f3b62f8 tf_resource_type=pagerduty_escalation_policy @caller=github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/internal/diag/diagnostics.go:62 diagnostic_detail="" diagnostic_severity=ERROR tf_proto_version=5.4 tf_rpc=ApplyResourceChange timestamp="2024-02-15T13:11:54.258+0100"
2024-02-15T13:11:54.269+0100 [TRACE] provider.terraform-provider-pagerduty_v3.7.1: Served request: tf_req_id=bf5f7fd9-6e12-dd97-4231-b2f04f3b62f8 tf_rpc=ApplyResourceChange tf_proto_version=5.4 tf_provider_addr=registry.terraform.io/pagerduty/pagerduty tf_resource_type=pagerduty_escalation_policy @caller=github.com/hashicorp/terraform-plugin-go@v0.20.0/tfprotov5/t
```

### Expected Behavior

I expected the escalation policy to update the target schedule.

### Actual Behavior

The provider is timed out after 5m with `403 Forbidden` error:

```bash
│ Error: PUT API call to https://api.pagerduty.com/escalation_policies/<omitted> failed: 403 Forbidden
│
│   with pagerduty_escalation_policy.team_oncall_escalation_policy,
│   on ../../../modules/pagerduty/pd-on-call-team/escalation_policy.tf line 1, in resource "pagerduty_escalation_policy" "team_oncall_escalation_policy":
│    1: resource "pagerduty_escalation_policy" "team_oncall_escalation_policy" {
```

### Steps to Reproduce

Just `terraform apply`.

### Important Factoids

We tried:
1. Different PagerDuty provider versions -- :x: 
2. Different Terraform versions -- :x: 
3. Hardcoding PagerDuty API token in the provider authentication block.
4. New PagerDuty API token with same permissions -- :x: 
5. Increasing provider logs verbosity, this times out after 5 minutes and emits same error as above w/o specific details -- :x: 
6. Updating escalation policy with original PagerDuty token using Python client -- :white_check_mark: 

Re. (6), example snippet:

```python
#!/usr/bin/env python

import os
from pdpyras import APISession
from pprint import pprint as pp


def get_api_token():
    """Retrieve API token from environment variables."""
    token = os.getenv("PAGERDUTY_API_TOKEN")
    if not token:
        raise ValueError("Please set the PAGERDUTY_API_TOKEN environment variable.")
    return token


def update_escalation_policy_target(session, policy_id, target_id, new_target_details):
    """Update the target in the escalation policy with new details."""
    policy = session.rget(f"/escalation_policies/{policy_id}")
    pp(policy)

    policy_updated = False
    del policy["teams"]  # remove teams as we don't use them and have not support plan

    for rule in policy.get("escalation_rules", []):
        for target in rule.get("targets", []):
            if target["id"] == target_id:
                target.update(new_target_details)
                policy_updated = True
                break 

    return policy, policy_updated


def main():
    api_token = get_api_token()
    session = APISession(api_token)

    escalation_policy_id = "PJ81HJ6"
    target_id_to_update = "PZ4PJ9I"

    new_target_details = {
        "id": "PAIH89Y",
        "self": "https://api.pagerduty.com/schedules/PAIH89Y",
        "summary": "<omitted>",
        "html_url": "https://<companyname>.pagerduty.com/schedules/PAIH89Y",
    }

    try:
        updated_policy, updated = update_escalation_policy_target(
            session, escalation_policy_id, target_id_to_update, new_target_details
        )

        if updated:
            session.rput(
                f"/escalation_policies/{escalation_policy_id}", json=updated_policy
            )
            print(f"Updated escalation policy {escalation_policy_id} successfully.")
        else:
            print(
                f"No matching rule found for update in policy {escalation_policy_id}."
            )

    except Exception as e:
        error_message = str(e)
        print(f"Error updating escalation policy: {error_message}")


if __name__ == "__main__":
    main()
```

--- a/pagerduty/resource_pagerduty_escalation_policy.go
+++ b/pagerduty/resource_pagerduty_escalation_policy.go
@@ -172,10 +172,23 @@ func fetchEscalationPolicy(d *schema.ResourceData, meta interface{}, errCallback
 
 	o := &pagerduty.GetEscalationPolicyOptions{Includes: []string{"escalation_rule_assignment_strategies"}}
 
+	var escalationPolicyFirstAttempt *pagerduty.EscalationPolicy
+
+	escalationPolicyFirstAttempt, _, err = client.EscalationPolicies.Get(d.Id(), o)
+	if err != nil && isErrCode(err, http.StatusForbidden) {
+		// Removing the inclusion of escalation_rule_assignment_strategies for
+		// accounts wihtout the required entitlements.
+		o = nil
+	}
+
+	if err == nil && escalationPolicyFirstAttempt == nil {
+		return setResourceEPProps(d, escalationPolicyFirstAttempt)
+	}
+
 	return retry.Retry(5*time.Minute, func() *retry.RetryError {
 		escalationPolicy, _, err := client.EscalationPolicies.Get(d.Id(), o)
 		if err != nil {
-			if isErrCode(err, http.StatusBadRequest) {
+			if isErrCode(err, http.StatusBadRequest) || isErrCode(err, http.StatusForbidden) {
 				return retry.NonRetryableError(err)
 			}
 
@@ -189,20 +202,28 @@ func fetchEscalationPolicy(d *schema.ResourceData, meta interface{}, errCallback
 			return nil
 		}
 
-		d.Set("name", escalationPolicy.Name)
-		d.Set("description", escalationPolicy.Description)
-		d.Set("num_loops", escalationPolicy.NumLoops)
-
-		if err := d.Set("teams", flattenTeams(escalationPolicy.Teams)); err != nil {
-			return retry.NonRetryableError(fmt.Errorf("error setting teams: %s", err))
-		}
-
-		if err := d.Set("rule", flattenEscalationRules(escalationPolicy.EscalationRules)); err != nil {
+		err = setResourceEPProps(d, escalationPolicy)
+		if err != nil {
 			return retry.NonRetryableError(err)
 		}
 
 		return nil
 	})
+}
+
+func setResourceEPProps(d *schema.ResourceData, escalationPolicy *pagerduty.EscalationPolicy) error {
+	d.Set("name", escalationPolicy.Name)
+	d.Set("description", escalationPolicy.Description)
+	d.Set("num_loops", escalationPolicy.NumLoops)
+
+	if err := d.Set("teams", flattenTeams(escalationPolicy.Teams)); err != nil {
+		return fmt.Errorf("error setting teams: %s", err)
+	}
+
+	if err := d.Set("rule", flattenEscalationRules(escalationPolicy.EscalationRules)); err != nil {
+		return err
+	}
+	return nil
 }
 
 func resourcePagerDutyEscalationPolicyUpdate(d *schema.ResourceData, meta interface{}) error {

